### PR TITLE
explain #176 in packet_device doc

### PR DESCRIPTION
**Pull request by t0mk**
This PR adds doc to packet_device, hardware_reservation_id, next-available use-case. It's the outcome of discussion in #176.

### CODE REVIEW

**Comment by t0mk** [comment]
@rata Please check the wording, and suggest improvements if you feel like :)

File: website/docs/r/device.html.markdown [hunk 1]
```diff
@@ -134,6 +134,7 @@ The following arguments are supported:
 * `always_pxe` (Optional) - If true, a device with OS `custom_ipxe` will
   continue to boot via iPXE on reboots.
 * `hardware_reservation_id` (Optional) - The `full ID` of the hardware reservation where you want this device deployed, or `next-available` if you want to pick your next available reservation automatically.
+  Please be careful if using hw reservation UUID and `next-available` together for the same pool of resevations. It might happen that the reservation which Packet API will pick as next-available is the reservation which you refer with UUID in another packet_device resource. To fix this, have the `next-available` resource  [explicitly depend_on](https://learn.hashicorp.com/terraform/getting-started/dependencies.html#implicit-and-explicit-dependencies) the resource with hw reservation UUID, so that the latter is created first.
```
Comment (rata) [inline website/docs/r/device.html.markdown]: s/To fix this/To workaround this/ ? This is personal taste, both are totally fine IMHO

Comment (rata) [inline website/docs/r/device.html.markdown]: Maybe it makes sense to add something like "For more detailed information, see issue #176"?

[the remark this answers is not in the document]
Comment (t0mk) [reply website/docs/r/device.html.markdown]: OK

Comment (t0mk) [reply website/docs/r/device.html.markdown]: I will

Comment (t0mk) [reply website/docs/r/device.html.markdown]: I agree

**Comment by rata** [comment]
LBTM (Looks even Better To Me :-P)

**Pull request merged**